packages/admin-ui/src/Switch/Switch.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import * as SwitchPrimitives from "@radix-ui/react-switch";
+import { makeDecoratable } from "@webiny/react-composition";
+import { cva, type VariantProps } from "class-variance-authority";
+import { cn, generateId } from "~/utils";
+import { Label } from "~/Label";
+
+const switchVariants = cva("flex items-center space-x-2", {
+    variants: {
+        labelPosition: {
+            start: "",
+            end: "flex-row-reverse space-x-reverse "
+        }
+    },
+    defaultVariants: {
+        labelPosition: "start"
+    }
+});
+
+interface SwitchProps
+    extends React.ComponentPropsWithoutRef<typeof SwitchPrimitives.Root>,
+        VariantProps<typeof switchVariants> {
+    label: React.ReactNode;
+}
+
+const SwitchBase = React.forwardRef<React.ElementRef<typeof SwitchPrimitives.Root>, SwitchProps>(
+    ({ id: initialId, className, label, labelPosition, ...props }, ref) => {
+        const id = generateId(initialId);
+
+        return (
+            <div className={cn(switchVariants({ labelPosition, className }))}>
+                {label && <Label text={label} htmlFor={id} />}
+                <SwitchPrimitives.Root
+                    className={cn(
+                        "peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
+                    )}
+                    {...props}
+                    ref={ref}
+                    id={id}
+                    aria-label={label ? String(label) : "Switch"}
```

Comment:
I can see that label can be string, but also `React.ReactNode`. So, `aria-label` would be set incorrectly if I've sent something like this:
![image](https://github.com/user-attachments/assets/bbf8756e-02ff-4c03-acf8-0f52061aa828)

Maybe we just do `typeof label === 'string' ? label || "Switch"`

Reply:
You are right, thank you!